### PR TITLE
COM-410: Use new field components

### PR DESCRIPTION
**Pull request by juliawegmayr**
Use new field components to simplify forms. 

New available field components: 

-   TextField
-   TextAreaField
-   SearchField
-   SelectField
-   CheckboxField
-   SwitchField
-   ColorField
-   DateField
-   DateRangeField
-   TimeField
-   TimeRangeField
-   DateTimeField

### CODE REVIEW

File: demo/admin/src/common/ComponentDemo.tsx [hunk 1]
```diff
@@ -224,24 +202,18 @@ export function ComponentDemo(): React.ReactElement {
...
                             </FieldContainer>
 
                             <Field name="switch" label="Switch with label">
                                 {(props) => <FormControlLabel label={null} control={<FinalFormSwitch {...props} />} />}
                             </Field>
 
-                            <Field name="switch">
-                                {(props) => <FormControlLabel label="Switch with inline label" control={<FinalFormSwitch {...props} />} />}
-                            </Field>
+                            <SwitchField name="switch" fieldLabel="Switch with inline label" />
```
Comment (jamesricky) [inline demo/admin/src/common/ComponentDemo.tsx:214]: Shouldn't the `SwitchField` with the `fieldLabel` replace the `Field` above and have that field's label as well?

File: demo/admin/src/common/ComponentDemo.tsx [hunk 2]
```diff
@@ -183,32 +169,24 @@ export function ComponentDemo(): React.ReactElement {
...
+                                </MenuItem>
+                                <MenuItem value="Option 2">
+                                    <CustomSelectItem icon={<Snips />} primary="Option 2" secondary="Secondary text" />
+                                </MenuItem>
+                                <MenuItem value="Option 3">
+                                    <CustomSelectItem icon={<Snips />} primary="Option 3" secondary="Secondary text" />
+                                </MenuItem>
+                            </SelectField>
+
+                            <TextAreaField name="textArea" label="Text Area" multiline minRows={3} fullWidth />
```
Comment (jamesricky) [inline demo/admin/src/common/ComponentDemo.tsx:189]: `multiline` (and possibly `minRows`) should not be necessary, when using `TextAreaField`

File: demo/admin/src/common/ComponentDemo.tsx [hunk 3]
```diff
@@ -224,24 +202,18 @@ export function ComponentDemo(): React.ReactElement {
...
-                                </Field>
-                                <Field name="multiple-choice-2" type="checkbox" fullWidth>
-                                    {(props) => <FormControlLabel label="Option 2" control={<FinalFormCheckbox {...props} />} />}
-                                </Field>
-                                <Field name="multiple-choice-3" type="checkbox" fullWidth>
-                                    {(props) => <FormControlLabel label="Option 3" control={<FinalFormCheckbox {...props} />} />}
-                                </Field>
+                                <CheckboxField name="multiple-choice-1" type="checkbox" label="Option 1" fullWidth />
+                                <CheckboxField name="multiple-choice-2" type="checkbox" label="Option 2" fullWidth />
+                                <CheckboxField name="multiple-choice-3" type="checkbox" label="Option 3" fullWidth />
```
Comment (jamesricky) [inline demo/admin/src/common/ComponentDemo.tsx:207]: Isn't `type="checkbox"` already defined in `CheckboxField`? 🤔

File: demo/admin/src/products/ProductForm.tsx [hunk 4]
```diff
@@ -236,21 +219,18 @@ function ProductForm({ id }: FormProps): React.ReactElement {
...
-                                <FormControlLabel
-                                    label={<FormattedMessage id="product.inStock" defaultMessage="In stock" />}
-                                    control={<FinalFormCheckbox {...props} />}
-                                />
-                            )}
-                        </Field>
+                        <CheckboxField
+                            name="inStock"
+                            label={<FormattedMessage id="product.inStock" defaultMessage="In stock" />}
+                            type="checkbox"
```
Comment (jamesricky) [inline demo/admin/src/products/ProductForm.tsx:231]: Isn't type="checkbox" already defined in CheckboxField? 🤔

Comment (jamesricky) [reply demo/admin/src/products/ProductForm.tsx:231]:
```suggestion
```

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Comment by johnnyomair** [comment]
@juliawegmayr please resolve conflicts, thanks!

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**